youtube_dl/options.py

```diff
@@ -848,19 +854,31 @@ def compat_conf(conf):
             return conf
 
         command_line_conf = compat_conf(sys.argv[1:])
+        opts, args = parser.parse_args(command_line_conf)
 
         if '--ignore-config' in command_line_conf:
             system_conf = []
             user_conf = []
+        elif '--configfile' in command_line_conf:
+            if not os.path.isfile(opts.configfile):
+                write_string('configfile not found, using default configfiles instead \n')
+                system_conf = _readOptions('/etc/youtube-dl.conf')
+                user_conf = _readUserConf()
+            else:
+                user_conf = []
+                system_conf = _readOptions(opts.configfile)
```

Comment:
Config files specified via the command line sounds like a user configuration for me.